### ISSUE TITLE
"Pure" vs "Impure" Interfaces

### ISSUE THREAD

**Issue by szeying02** [nus-cs2030/2122-s2#26]
# Description

What is the difference between the "Pure" and "Impure" interfaces, and what are they? This is mentioned in the footnote of slide 7 in lecture 5. Thank you!

## Topic:

Interface

**Comment by hwkchia** [comment]
Yes, they are mentioned in the slides for completeness.. but out of scope in our module.  They are just interfaces with some methods defined... something like abstract classes, but as they are interfaces they do not face the "inherit from one parent" constraint.  GIY: java interfaces backward compatibility
